daffodil-lib/src/main/scala/org/apache/daffodil/lib/exceptions/SchemaFileLocatable.scala

```diff
@@ -35,24 +32,42 @@ trait HasSchemaFileLocation extends LookupLocation {
 }
 
 object SchemaFileLocation {
-  def apply(context: SchemaFileLocatable, tunables: DaffodilTunables) =
+  def apply(context: SchemaFileLocatable) =
     new SchemaFileLocation(
       context.lineNumber,
       context.columnNumber,
       context.uriString,
+      context.diagnosticFilepath,
       context.toString,
       context.diagnosticDebugName,
-      tunables.maxParentDirectoriesForDiagnostics,
     )
+
+  def apply(
+    lineNumber: Option[String],
+    columnNumber: Option[String],
+    uriString: String,
+    diagnosticFilepath: String,
+    contextToString: String,
+    diagnosticDebugName: String,
+  ) = {
+    new SchemaFileLocation(
+      lineNumber,
+      columnNumber,
+      uriString,
+      diagnosticFilepath,
+      contextToString,
+      diagnosticDebugName,
+    )
+  }
```

Comment:
This essentially just makes the `SchemaFileLocation` constructor public. If we wanted to keep it that way we could create a custom `XercesSchmeaFileLocatable`. Then the places where we use this new constructor, we instead do something like
```scala
val xsfl = new XerexSchemaFileLocatable(xercesError, ....)
val schemaLocation = SchemaLocation(xsfl)
new SchemaDefinitionError(schemaLocation, ...)
```
```

Reply:
I think your comment is already resolved, Steve.  This PR defines a new custom XercesSchemaFileLocatable and passes it to this new apply() constructor.  The places where the apply() constructor is called already looks very similar to your code except for the addition of a second parameter.  The second parameter looks important and shouldn't be omitted.

```scala
    val xsfl = new XercesSchemaFileLocatable(err, "")
    val newSchemaLocation = SchemaFileLocation(
      xsfl,
      schemaFileLocation,
    )
    new SchemaDefinitionError(
      newSchemaLocation,
      "Error loading schema due to %s",
      errMessage,
    )
```